vnext/Shared/Modules/BlobModule.cpp

```diff
@@ -0,0 +1,538 @@
+         if (auto prop = propBag.Get(propId)) {
+           wsProxy = prop.Value().lock();
+         }
+         if (!wsProxy) {
+           return;
+         }
+
+         auto blob = jsArgAsObject(args, 0);
+         auto blobId = blob[blobIdKey].getString();
+         auto offset = blob[offsetKey].getInt();
+         auto size = blob[sizeKey].getInt();
+         auto socketID = jsArgAsInt(args, 1);
+
+         auto data = persistor->ResolveMessage(std::move(blobId), offset, size);
+
+         auto buffer = CryptographicBuffer::CreateFromByteArray(data);
+         auto winrtString = CryptographicBuffer::EncodeToBase64String(std::move(buffer));
+         auto base64String = Common::Unicode::Utf16ToUtf8(std::move(winrtString));
+
+         wsProxy->SendBinary(std::move(base64String), socketID);
+       }},
+
+      {"createFromParts",
+       // As of React Native 0.67, instance is set AFTER CxxModule::getMethods() is invoked.
+       // Use getInstance() directly once
+       // https://github.com/facebook/react-native/commit/1d45b20b6c6ba66df0485cdb9be36463d96cf182 becomes available.
+       [persistor = m_blobPersistor, weakState = weak_ptr<SharedState>(m_sharedState)](dynamic args) {
+         auto parts = jsArgAsArray(args, 0); // Array<Object>
+         auto blobId = jsArgAsString(args, 1);
+         vector<uint8_t> buffer{};
+
+         for (const auto &part : parts) {
+           auto type = part[typeKey].asString();
+           if (blobKey == type) {
+             auto blob = part[dataKey];
+             auto bufferPart =
+                 persistor->ResolveMessage(blob[blobIdKey].asString(), blob[offsetKey].asInt(), blob[sizeKey].asInt());
+
+             buffer.reserve(buffer.size() + bufferPart.size());
+             buffer.insert(buffer.end(), bufferPart.begin(), bufferPart.end());
+           } else if ("string" == type) {
+             auto data = part[dataKey].asString();
+
+             buffer.reserve(buffer.size() + data.size());
+             buffer.insert(buffer.end(), data.begin(), data.end());
+           } else {
+             if (auto state = weakState.lock()) {
+               auto message = "Invalid type for blob: " + type;
+               Modules::SendEvent(state->Module->getInstance(), "blobFailed", std::move(message));
+             }
+             return;
+           }
+         }
+
+         persistor->StoreMessage(std::move(buffer), std::move(blobId));
+       }},
+
+      {"release",
+       [persistor = m_blobPersistor](dynamic args) // blobId: string
+       {
+         auto blobId = jsArgAsString(args, 0);
+
+         persistor->RemoveMessage(std::move(blobId));
+       }}};
+}
+
+#pragma endregion CxxModule
+
+#pragma endregion BlobModule
+
+#pragma region MemoryBlobPersistor
+
+#pragma region IBlobPersistor
+
+winrt::array_view<uint8_t> MemoryBlobPersistor::ResolveMessage(string &&blobId, int64_t offset, int64_t size) noexcept {
+  scoped_lock lock{m_mutex};
+
+  auto &data = m_blobs.at(std::move(blobId));
```

Comment:
>at [](http://example.com/codeflow?start=23&length=2)

What if it is not found? It will cause the app crash, right?

Reply:
True. Solved by resorting to the iterator.